tests/integration-tests/conftest.py

```diff
@@ -1160,17 +1160,13 @@ def placement_group_stack(cfn_stacks_factory, request, region):
 
 
 @pytest.fixture(scope="class")
-def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_stack):
+def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_stack: CfnVpcStack):
     logging.info("Setting up the ODCR stack")
     odcr_template = Template()
     odcr_template.set_version()
     odcr_template.set_description("ODCR stack to test open, targeted, and PG ODCRs")
-    availability_zone = (
-        boto3.resource("ec2").Subnet(get_vpc_snakecase_value(vpc_stack)["public_subnet_id"]).availability_zone
-    )
-    availability_zone_2 = (
-        boto3.resource("ec2").Subnet(get_vpc_snakecase_value(vpc_stack)["public_az2_subnet_id"]).availability_zone
-    )
+    availability_zone = boto3.resource("ec2").Subnet(vpc_stack.get_all_public_subnets()[0]).availability_zone
+    availability_zone_2 = boto3.resource("ec2").Subnet(vpc_stack.get_all_public_subnets()[1]).availability_zone
```

Comment:
you can call `get_all_public_subnets` only one time to save 1 boto3 call to retrieve cfn outputs.